src/app/errorClasses.ts

```diff
@@ -1,21 +1,25 @@
 export class DatabaseError extends Error {
-    constructor(errorCategory: DatabaseOperation, faultyArea = "", logId: string) {
+    constructor(errorCategory: DatabaseOperation, faultyArea = "", logId?: string) {
         super(
             `An error occurred completing the ${errorCategory} operation` +
                 (faultyArea !== "" ? ` (${faultyArea})` : "") +
                 ". Please try again." +
-                `Log ID: ${logId}`
+                logId
+                ? `Log ID: ${logId}`
```

Comment:
(same in the other places)

```suggestion
                ? ` Log ID: ${logId}`
```